database/interface.py

```diff
@@ -190,113 +187,15 @@ def update_harvest_source(self, source_id, updates):
-                    self.db.query(HarvestRecord.id).filter_by(
-                        harvest_source_id=source_id
-                    )
-                )
-            ).delete(synchronize_session=False)
-            self.db.query(HarvestRecord).filter_by(harvest_source_id=source_id).delete()
-            self.db.commit()
-
-        source = self.db.get(HarvestSource, source_id)
-        if source is None:
-            return "Harvest source not found"
-
-        organization_id = source.organization_id
-
-        records = (
-            self.db.query(HarvestRecord).filter_by(harvest_source_id=source_id).all()
-        )
-
-        if not records:
-            return "Harvest source has no records to clear."
-
-        ckan_ids = [record.ckan_id for record in records if record.ckan_id is not None]
-        error_records = [record for record in records if record.status == "error"]
-        jobs_in_progress = self.get_all_harvest_jobs_by_filter(
-            {"harvest_source_id": source.id, "status": "in_progress"}
-        )
-
-        # Ensure no jobs are in progress
-        if jobs_in_progress:
-            return (
-                "Error: A harvest job is currently in progress. "
-                "Cannot clear datasets."
-            )
-
-        # Ensure (error_records + ckan_ids) = total records
-        if len(error_records) + len(ckan_ids) != len(records):
-            return (
-                "Error: Not all records are either in an error state "
-                "or have a CKAN ID. Cannot proceed without clearing the dataset."
-            )
-
-        if not ckan_ids:
-            _clear_harvest_records()
-            return "Harvest source cleared successfully."
-
-        ckan = RemoteCKAN(os.getenv("CKAN_API_URL"), apikey=os.getenv("CKAN_API_TOKEN"))
-
-        result = ckan.action.package_search(fq=f"harvest_source_id:{source_id}")
-        ckan_datasets = result["count"]
-        start = datetime.now(timezone.utc)
-        retry_count = 0
-        retry_max = 20
-
-        # Retry loop to handle timeouts from cloud.gov and CKAN's Solr backend,
-        # ensuring datasets are cleared despite possible interruptions.
-        while ckan_datasets > 0 and retry_count < retry_max:
-            result = ckan.action.package_search(fq=f"harvest_source_id:{source_id}")
-            ckan_datasets = result["count"]
-            logger.info(
-                f"Attempt {retry_count + 1}: "
-                f"{ckan_datasets} datasets remaining in CKAN"
-            )
-            try:
-                ckan.action.bulk_update_delete(
-                    datasets=ckan_ids, org_id=organization_id
-                )
-            except ckanapi.errors.CKANAPIError as api_err:
-                logger.error(f"CKAN API error: {api_err}")
-            except Exception as err:
-                logger.error(f"Error occurred: {err} \n error_type: {type(err)}")
-                return f"Error occurred: {err}"
```

Comment:
I'm not quite following; where does this logic move to? Are we losing the ability to delete from CKAN, or is this cleanup of old code?

Reply:
I believe we just trick the harvester into doing the work for us by pushing an empty list as incoming harvest object.

Reply:
Yes, we removed the separate call to the CKAN API for deleting/purging. Instead, we now reuse the current harvest process with an empty list, which effectively purges all datasets from both CKAN and the harvestDB.